File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/03-additional/07.safePasswordsGenerator.js
function safePasswordsGenerator(input) {

    let a = Number(input[0]);
    let b = Number(input[1]);
    let capacity = Number(input[2]);

    let printline = '';
    let passCounter = 0;
    let isBreak = false;

    for (let i = 35; i <= 55; i++) {
        for (let j = 64; j <= 96; j++) {
            for (let x = 1; x <= a; x++) {
                for (let y = 1; y <= b; y++) {
                    printline += `${String.fromCharCode(i)}${String.fromCharCode(j)}${x}${y}${String.fromCharCode(j)}${String.fromCharCode(i)}|`;
                    passCounter++;
                    i++;
                    j++;
                    if (passCounter == capacity) {
                        break;
                    }
                    if (i > 55) {
                        i = 35;
                    }
                    if (j > 96) {
                        j = 64;
                    }
                }
                if (passCounter == capacity) {
                    break;
                }
                if (x == a) {
                    isBreak = true;
                    break;
                }
            }
            if (passCounter == capacity) {
                break;
            }
            if (isBreak) {
                break;
            }
        }
        if (passCounter == capacity) {
            break;
        }
        if (isBreak) {
            break;
        }
    }

    console.log(printline);
}
safePasswordsGenerator(['20', '50', '10']);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/03.additional/07.safe_passwords_generator.py
a = int(input())
b = int(input())
capacity = int(input())

print_line = ''
pass_counter = 0
is_break = False

for i in range(35, 56):
    for j in range(64, 97):
        for x in range(1, a + 1):
            for y in range(1, b + 1):
                print_line += f'{str(chr(i))}{str(chr(j))}{x}{y}{str(chr(j))}{str(chr(i))}|'
                pass_counter += 1
                i += 1
                j += 1
                if pass_counter == capacity:
                    break         
                if i > 55:
                    i = 35
                if j > 96:
                    j = 64
            if pass_counter == capacity:
                break
            if x == a:
                is_break = True
                break 
        if pass_counter == capacity:
            break
        if is_break:
            break
    if pass_counter == capacity:
        break
    if is_break:
        break

print(print_line)
